Wait for Release - 48911: Application Icon Preview (#375)
replicatedhq/replicated-docs: docs/vendor/admin-console-customize-app-icon.md

To add a custom application icon:

1. In the [vendor portal](https://vendor.replicated.com/apps), click **Releases**. Click **Create release** to create a new release, or click **Edit YAML** to edit an existing release.
1. Create or open the Application custom resource manifest file. An Application custom resource manifest file has `apiVersion: kots.io/v1beta1` and `kind: Application`.

1. In the preview section of the Help pane:

    1. If your Application manifest file is already populated with an `icon` key, the icon displays in the preview. Click **Preview a different icon** to access the preview options.

    1. Drag and drop an icon image file to the drop zone. Alternatively, paste a link or Base64 encoded data URL in the text box. Click **Preview**.

      ![Application icon preview](/images/app-icon-preview.png)

>     1. (Air gap only) If you paste a link to the image in the text box, click **Preview** and **Base64 encode icon** to convert the image to Base64 encoded data URL. An encoded URL displays that you can copy and paste into the Application manifest. Base64 encoding is required for images used with air gap installations.

Reviewer: ```suggestion
    1. (Air gap only) If you paste a link to the image in the text box, click **Preview** and **Base64 encode icon** to convert the image to a Base64 encoded data URL. An encoded URL displays that you can copy and paste into the Application manifest. Base64 encoding is required for images used with air gap installations.
```

1. In the [vendor portal](https://vendor.replicated.com/apps), click **Releases**. Click **Create release** to create a new release, or click **Edit YAML** to edit an existing release.

> 1. Create or open the Application custom resource manifest file. An Application custom resource manifest file has `kind: Application`.

Reviewer: We also support another file that is `kind: Application`, but the API version is `app.k8s.io/v1beta1`.
```suggestion
1. Create or open the Application custom resource manifest file. An Application custom resource manifest file has `apiVersion: kots.io/v1beta1` and `kind: Application`.
```

> 1. In the Help pane, click **Show** next to **Application icon preview**.

Reviewer: Star made a change so that it is expanded by default, so you don't have to click **Show** first now.

  The preview section expands. If your Application manifest file is already populated with an `icon` key, the icon displays in the preview. Otherwise, the preview is blank.

>     1. Drag and drop an icon image file to the drop zone. Alternatively, paste a link or Base64 encoded string in the text box. Click **Preview** or **Preview a different image**. Which button displays depends on whether a preview already exists.

Author: Thanks!

>     1. Drag and drop an icon image file to the drop zone. Alternatively, paste a link or Base64 encoded data URL in the text box. Click **Preview** or **Preview a different icon**. Which button displays depends on whether a preview already exists.

Reviewer: If an icon is already present and previewed, you have to click "Preview a different icon" first before you can select an image or paste a link. So this needs to be reworded a bit to reflect that you do that first if an icon is already present, and then the options are basically the same as if an icon wasn't present (drag and drop or paste link). Ultimately then you should click "Preview" either way in order to see the preview.

    ![Application icon preview](/images/app-icon-preview.png)

>     1. (Air Gap only) If you drop or paste a non-encoded image into the fields, click **Base64 encode image** to convert the image to Base64. An encoded URI displays that you can copy and paste into the Application manifest. Base64 encoding is required for images used with air gap installations.

Reviewer 2: Same with this button you mention here. I will change the text to say "Base64 encode icon".

>     1. (Air Gap only) If you drop or paste a non-encoded image into the fields, click **Base64 encode icon** to convert the image to Base64. An encoded URI displays that you can copy and paste into the Application manifest. Base64 encoding is required for images used with air gap installations.

Reviewer: Not sure about this, but just in case.
```suggestion
    1. (Air gap only) If you drop or paste a non-encoded image into the fields, click **Base64 encode icon** to convert the image to Base64. An encoded URI displays that you can copy and paste into the Application manifest. Base64 encoding is required for images used with air gap installations.
```

Author: I wasn't sure either. Change accepted :)

>     1. (Air Gap only) If you drop or paste a non-encoded image into the fields, click **Base64 encode icon** to convert the image to Base64. An encoded URI displays that you can copy and paste into the Application manifest. Base64 encoding is required for images used with air gap installations.

Reviewer: The **Base64 encode icon** button only appears if you paste a link. If you drag and drop an icon or if you paste a data URL, it will do the encoding automatically. It only asks if you want to do it if you use a link.

Also, I don't know if the wording is perfect, but I added a mention of **Preview** since you have to click that after pasting the link before you can click the base64 button. Up to you if you think that's clear already.
```suggestion
    1. (Air Gap only) If you paste a link to the image into the field and click **Preview**, click **Base64 encode icon** to convert the image to a Base64 encoded data URL. You can copy and paste the data URL into the Application manifest. Base64 encoding is required for images used with air gap installations.
```

Author: Hmm, I think this is not entirely clear either. let's see what you think of my blend of our edits.I moved the note up above the screenshot to make the exceptions more prominent.

    ![Base64 encode image button](/images/app-icon-preview-base64.png)

    :::note

>     If you pasted a Base64 string into the text box, the **Base64 encode icon** button does not display because the image is already encoded.

Reviewer: ```suggestion
    If you pasted a Base64 encoded data URL into the text box, the **Base64 encode icon** button does not display because the image is already encoded.
```

    :::

    1. Click **Preview a different icon** to preview a different icon if needed.

> 1. In the Application manifest, under `spec`, add an `icon` key that includes a link or the Base64 URI to the desired image.

Reviewer 2: We may want to use the "Base64 encoded data URL" language here as well
